### PR TITLE
feat: support checking all blocks in material add block panel

### PR DESCRIPTION
**Pull request by gene9831**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Background and solution

物料插件-区块分组-添加区块，区块列表没有全选功能

改动点：
- 给区块列表添加全选功能
- 抽取了一个 `SelectAll` 组件到 `common` 包

### What is the new behavior?

image

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Release Notes

- **New Features**
  - Introduced a `Select All` component for bulk selection in the block list.
  - Enhanced block selection functionality with new methods for selecting and deselecting all items.

- **Bug Fixes**
  - Improved error handling in the `addBlocks` method to ensure selection state resets after operations.

- **Style**
  - Updated styling for checkboxes and various components to enhance visual representation and user experience.

- **Documentation**
  - Updated event emissions for better interaction management within the block list components.

### CODE REVIEW

File: packages/common/component/SelectAll.vue [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<template>
+  <tiny-checkbox class="block-select-all" :indeterminate="isIndeterminate" v-model="selectedAll">
+    {{ hiddenLabel ? '' : '全选' }}
+  </tiny-checkbox>
+</template>
+
+<script setup lang="ts">
```
Comment (chilingling) [inline packages/common/component/SelectAll.vue]: 这里为啥要 `lang="ts"` ？

File: packages/common/component/SelectAll.vue [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    type: Array,
+    default: () => []
+  },
+  hiddenLabel: {
+    type: Boolean,
+    default: false
+  }
+})
+
+const emit = defineEmits(['selectAll', 'deselectAll'])
```
Comment (chilingling) [inline packages/common/component/SelectAll.vue]: 这里两个 `emit` 事件是不是变成一个  `change` 事件就好了？change 传值为 `true` 代表全选，传值为 `false` ，代表全不选

File: packages/common/component/SelectAll.vue [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    {{ hiddenLabel ? '' : '全选' }}
+  </tiny-checkbox>
+</template>
+
+<script setup lang="ts">
+import { Checkbox as TinyCheckbox } from '@opentiny/vue'
+import { computed, defineEmits, defineProps } from 'vue'
+
+const props = defineProps({
+  items: {
```
Comment (chilingling) [inline packages/common/component/SelectAll.vue]: 这里 `items` 是不是可以语义化一点，叫 `fullList` 或者 `originList` ？

File: packages/common/component/PluginBlockList.vue [hunk 4]
```diff
@@ -1,6 +1,14 @@
 <template>
   <div v-if="blockStyle === BlockStyles.Mini" class="header">
-    <div class="col-checkbox"></div>
+    <div class="col-checkbox" v-if="showCheckbox">
+      <select-all
+        :items="data"
+        :selected="checked"
+        :hidden-label="true"
+        @select-all="(items) => handleSelectAll(items)"
```
Comment (hexqi) [inline packages/common/component/PluginBlockList.vue]: `@select-all="(items) => handleSelectAll(items)"`可以简写成`@select-all=”handleSelectAll“`

File: packages/common/component/SelectAll.vue (hunk 2 above)
Comment (gene9831) [reply packages/common/component/SelectAll.vue]: 合并为一个`selectAll`，参数为数组时，则为全选，参数内容为传进来的 `allItems`；否则为全不选

File: packages/common/component/SelectAll.vue (hunk 3 above)
Comment (gene9831) [reply packages/common/component/SelectAll.vue]: 改为 `allItems`

File: packages/common/component/PluginBlockList.vue (hunk 4 above)
Comment (gene9831) [reply packages/common/component/PluginBlockList.vue]: 已修改

**Review by chilingling — APPROVED** [review]

**Pull request merged**